### PR TITLE
RenameTankPopup fully implemented.

### LINKED ISSUE #233: Rename Tanks
**Describe the solution you'd like**
Give the armory the ability to rename tanks.

**Describe alternatives you've considered**
Make tank names permanent. However their purpose can change, and destroying them would be meaningless.

**Additional context**
Need to make sure users cannot rename their tanks with a name that already exists.

### PR DESCRIPTION
**Pull request by blaiadam**
**Description**
Fully implemented RenameTankPopup. However, once the API rework is done, this will need changes.

Resolves #233 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
1. Go to armory.
2. Click Rename button on left-hand side.
3. Enter a new name (cannot be less than 3 or more than 20 characters).
4. View the change and your tank's new name.
5. Press f12 and check DB to make sure no errors occur.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/armory/SetWagerPopup.js [hunk 1]
```diff
@@ -58,16 +58,16 @@ class SetWagerPopup extends React.Component<Props, State> {
 
 		// Check if the wager was set. Close the popup if it is.
 		setWager(this.state.userWager, setSuccessful => {
-			if (setSuccessful) {
+			if (setSuccessful != null) {
```
Comment (SecondThread) [inline src/armory/SetWagerPopup.js]: err, isn't that a boolean?

File: src/armory/SetWagerPopup.js [hunk 2]
```diff
@@ -58,16 +58,16 @@ class SetWagerPopup extends React.Component<Props, State> {
...
 				this.setState({currentWager: this.state.userWager});
 			}
 			else {
 				this.setState({errorMessage: 'Could not set wager.'});
 			}
 		});
 		setFavoriteTankId(this.props.wagerTank._id, setSuccessful => {
-			if (setSuccessful) {
-				this.setState({setWagerOpen: false, currentWagerTank: this.props.wagerTank});
+			if (setSuccessful != null) {
```
Comment (SecondThread) [inline src/armory/SetWagerPopup.js]: ditto, should be a boolean, right?

**Review by SecondThread — APPROVED** [review]

File: src/armory/SetWagerPopup.js (hunk 1 above)
Comment (blaiadam) [reply src/armory/SetWagerPopup.js]: You asked me to change it to this in another PR.

Comment (SecondThread) [reply src/armory/SetWagerPopup.js]: oh, I could have been wrong lol. Is setSuccessful a boolean or an object? If it was a boolean I was definitely wrong. I thought you were doing a null check

Comment (SecondThread) [reply src/armory/SetWagerPopup.js]: Like if it is suppose to be a boolean just the if statement is fine

**Pull request merged**